pkg/controller/v1alpha2/applicationconfiguration/apply.go

```diff
@@ -95,7 +98,7 @@ func (a *workloads) Apply(ctx context.Context, status []v1alpha2.WorkloadStatus,
 				continue
 			}
 			t := trait.Object
-			if err := a.client.Apply(ctx, &trait.Object, ao...); err != nil {
+			if err := a.updatingClient.Apply(ctx, &trait.Object, ao...); err != nil {
```

Comment:
Why not change Apply() to update? That's more clear. Otherwise it still looks similar to `kubectl apply`.

Reply:
The primary concern is that if use client.Update(), we need more code, e.g., Get/Check-IfNotFound-Create/IfFound-SetResourceVersion-Update, just like what underlying of Apply() has done. So maybe we can just re-use Apply() here.

Reply:
Sounds good!